.github/workflows/ci.yml

```diff
@@ -26,37 +26,64 @@ jobs:
     name: Node.js v${{ matrix.NodeVersionDisplayName }} (${{ matrix.OS }})
     runs-on: ${{ matrix.OS }}
     steps:
+      - name: Create ~/.rush-user/settings.json
+        shell: pwsh
+        run: |
+          mkdir -p $HOME/.rush-user
+          @{ buildCacheFolder = Join-Path ${{ github.workspace }} rush-cache } | ConvertTo-Json > $HOME/.rush-user/settings.json
+
       - uses: actions/checkout@v3
         with:
           fetch-depth: 2
+          path: repo-a
 
       - name: Git config user
         run: |
           git config --local user.name "Rushbot"
           git config --local user.email "rushbot@users.noreply.github.com"
+        working-directory: repo-a
 
       - uses: actions/setup-node@v3
         with:
           node-version: ${{ matrix.NodeVersion }}
 
       - name: Verify Change Logs
         run: node common/scripts/install-run-rush.js change --verify
+        working-directory: repo-a
 
       - name: Rush Install
         run: node common/scripts/install-run-rush.js install
+        working-directory: repo-a
 
       # - if: runner.os == 'Linux'
       #   name: Start xvfb
       #   run: /usr/bin/Xvfb :99 -screen 0 1024x768x24 > /dev/null 2>&1 &
+      #   working-directory: repo-a
 
       - name: Rush retest (install-run-rush)
         run: node common/scripts/install-run-rush.js retest --verbose --production
+        working-directory: repo-a
 
       - name: Ensure repo README is up-to-date
         run: node repo-scripts/repo-toolbox/lib/start.js readme --verify
+        working-directory: repo-a
+
+      - name: Clone another copy of the repo to test the build cache
```

Comment:
Why do we need another copy of the repo?

Reply:
During the second `rush install` run, the `common/temp/<subspace>/pnpmfileSettings.json` files have a property (`"semverPath"`) that records where our generated pnpmfile can get the `semver` package. Running `rush install` on Rushstack with Rush in the current repo was causing an problem where the `common/temp/default/node_modules` folder is removed and recreated, so the `"semverPath"` points at a path that doesn't exist during the second subspace's installation.